third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/WMMA.cpp

```diff
@@ -217,62 +256,37 @@ Value generateWMMAOp(ConversionPatternRewriter &rewriter, Location loc,
                                tiedLower);
 }
 
-LinearLayout computeRepLayout(SmallVector<unsigned> &repA,
-                              SmallVector<unsigned> &repB,
-                              AMDWmmaEncodingAttr wmmaLayout, int K, int kDim,
-                              ArrayRef<int64_t> resShape, MLIRContext *ctx) {
-  StringAttr kRegister = StringAttr::get(ctx, "register");
-  StringAttr kLane = StringAttr::get(ctx, "lane");
-  StringAttr kWarp = StringAttr::get(ctx, "warp");
-  StringAttr kBlock = StringAttr::get(ctx, "block");
-
-  // To compute repetition layout, we need to divide WMMA layout
-  // with layout that describes single WMMA tile layout for all warps.
-  auto rank = resShape.size();
-  auto ctaLayout = wmmaLayout.getCtaLayout();
-  auto tile = wmmaLayout.getTileLayout(rank) * ctaLayout;
-
-  // Sometimes tile can be larger then the whole layout, depending on the
-  // block sizes so we need to adjust it in order to be sure that it is
-  // divisible.
-  llvm::SmallDenseMap<StringAttr, int64_t> shapeMap;
-  auto dims = standardOutDimNames(ctx, rank);
-  for (auto [dim, size] : llvm::zip(dims, resShape)) {
-    shapeMap[dim] = size;
-  }
-  tile = ensureLayoutNotLargerThan(tile, shapeMap);
-
-  auto wmmaLL = triton::gpu::toLinearLayout(resShape, wmmaLayout);
+static uint64_t packMN(uint32_t m, uint32_t n) {
+  return (uint64_t(m) << 32) | uint64_t(n);
+}
 
-  // Normalize layout so outputs are in the standard order.
-  wmmaLL = wmmaLL.transposeOuts(dims);
-  tile = tile.transposeOuts(dims);
-  auto repLayout = divideLeft(wmmaLL, tile).value();
+std::optional<int> findNextM(LinearLayout repLayout, int &reg, int elemsPerVec,
```

Comment:
This function looks overly complex. I get the feeling that there must be a simpler way to implement this logic by simply handling the linear layouts appropriately